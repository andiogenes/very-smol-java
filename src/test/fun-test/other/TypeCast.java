class Main {
    int bar() {
        return 9;
    }

    double foo() {
        return bar();
    }

    void main() {
        double a = foo() / 2;

        double b = 5;
        b = b / 2;

        // Вернет 2.0, т.к. 5/2 - целочисленное деление
        double c = 5 / 2;

        // Вернет 2.5, т.к. один из операндов - число с плавающей точкой.
        double d = 5.E0 / 2;

        // Вернет 2.5 по той же причине
        d = 5 / 2.E0;
    }
}