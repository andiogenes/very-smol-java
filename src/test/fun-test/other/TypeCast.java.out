a : DOUBLE = 4.5 : DOUBLE
b : DOUBLE = 5.0 : DOUBLE
b : DOUBLE = 2.5 : DOUBLE
c : DOUBLE = 2.0 : DOUBLE
d : DOUBLE = 2.5 : DOUBLE
d : DOUBLE = 2.5 : DOUBLE
